### PR TITLE
Merge masu openshift templates and Makefile, fixes #886

### LINKED ISSUE #886: Merge Masu OpenShift templates into Koku
## Objective
We want to move the OpenShift template files from masu to koku. 

## Acceptance Criteria

- [x] Masu is deployable from the koku repo via Makefile 
- [ ] The deployments mirror what exists currently for masu
- [ ] The hccm-build OpenShift project does not error

### PR DESCRIPTION
**Pull request by blentz**
This PR brings the Masu openshift templates over and integrates them into this repo. The Makefile is also updated to include commands for creating Masu's pods.

I've also done a bit of renaming and reorganizing as an attempt at integrate everything.

Here's a couple of screenshots of the results of running `make oc-create-all`:
Screenshot from 2019-06-12 12-13-09
Screenshot from 2019-06-12 12-12-02

And the CLI version:
```
oc get all
NAME                               READY     STATUS      RESTARTS   AGE
pod/koku-api-1-5xf9b               1/1       Running     0          36m
pod/koku-api-1-build               0/1       Completed   0          36m
pod/koku-celery-exporter-1-v5qc2   1/1       Running     0          13m
pod/koku-db-1-hdgm9                1/1       Running     0          36m
pod/koku-flower-1-build            0/1       Completed   0          11m
pod/koku-flower-1-deploy           0/1       Error       0          11m
pod/koku-listener-0                1/1       Running     0          13m
pod/koku-masu-flask-1-build        0/1       Completed   0          33m
pod/koku-masu-flask-1-deploy       0/1       Error       0          32m
pod/koku-masu-flask-2-bctkf        1/1       Running     0          17m
pod/koku-redis-1-wd24w             1/1       Running     0          36m
pod/koku-scheduler-1-9rjdc         1/1       Running     0          13m
pod/koku-worker-0                  1/1       Running     0          14m
pod/rabbitmq-0                     1/1       Running     0          29m
pod/rabbitmq-1-build               0/1       Completed   0          29m

NAME                                           DESIRED   CURRENT   READY     AGE
replicationcontroller/koku-api-1               1         1         1         36m
replicationcontroller/koku-celery-exporter-1   1         1         1         13m
replicationcontroller/koku-db-1                1         1         1         36m
replicationcontroller/koku-flower-1            0         0         0         11m
replicationcontroller/koku-masu-flask-1        0         0         0         32m
replicationcontroller/koku-masu-flask-2        1         1         1         17m
replicationcontroller/koku-redis-1             1         1         1         36m
replicationcontroller/koku-scheduler-1         1         1         1         13m

NAME                           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                AGE
service/koku-api               ClusterIP   172.30.62.189    <none>        8080/TCP                               36m
service/koku-celery-exporter   ClusterIP   172.30.56.200    <none>        8888/TCP                               13m
service/koku-db                ClusterIP   172.30.225.176   <none>        5432/TCP                               36m
service/koku-flask             ClusterIP   172.30.72.168    <none>        8080/TCP,9100/TCP                      33m
service/koku-flower            ClusterIP   172.30.242.29    <none>        5555/TCP                               11m
service/koku-listener-stats    ClusterIP   172.30.133.93    <none>        9999/TCP                               13m
service/koku-redis             ClusterIP   172.30.223.225   <none>        6379/TCP                               36m
service/masu-stats             ClusterIP   172.30.153.138   <none>        9999/TCP                               14m
service/rabbitmq               ClusterIP   None             <none>        4369/TCP,5671/TCP,5672/TCP,25672/TCP   29m

NAME                             DESIRED   CURRENT   AGE
statefulset.apps/koku-listener   1         1         13m
statefulset.apps/koku-worker     1         1         14m
statefulset.apps/rabbitmq        1         1         29m

NAME                                             REFERENCE                   TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
horizontalpodautoscaler.autoscaling/koku-api     DeploymentConfig/koku-api   <unknown>/80%   1         10        0          36m
horizontalpodautoscaler.autoscaling/koku-flask   DeploymentConfig/koku       <unknown>/90%   1         3         0          33m

NAME                                                      REVISION   DESIRED   CURRENT   TRIGGERED BY
deploymentconfig.apps.openshift.io/koku-api               1          1         1         config,image(koku-api:latest)
deploymentconfig.apps.openshift.io/koku-celery-exporter   1          1         1         config,image(koku-masu-flask:latest)
deploymentconfig.apps.openshift.io/koku-db                1          1         1         config,image(postgresql:9.6)
deploymentconfig.apps.openshift.io/koku-flower            1          1         0         config,image(koku-flower:latest)
deploymentconfig.apps.openshift.io/koku-masu-flask        2          1         1         config,image(koku-masu-flask:latest)
deploymentconfig.apps.openshift.io/koku-redis             1          1         1         image(redis:5.0.4)
deploymentconfig.apps.openshift.io/koku-scheduler         1          1         1         config,image(koku-masu-flask:latest)

NAME                                             TYPE      FROM         LATEST
buildconfig.build.openshift.io/koku-api          Docker    Git          1
buildconfig.build.openshift.io/koku-flower       Docker    Git          1
buildconfig.build.openshift.io/koku-masu-flask   Source    Git          1
buildconfig.build.openshift.io/rabbitmq          Docker    Git@master   1

NAME                                         TYPE      FROM          STATUS     STARTED          DURATION
build.build.openshift.io/koku-api-1          Docker    Git@a3fa4dc   Complete   36 minutes ago   7s
build.build.openshift.io/koku-masu-flask-1   Source    Git@6ae7621   Complete   33 minutes ago   1m8s
build.build.openshift.io/rabbitmq-1          Docker    Git@6ae7621   Complete   29 minutes ago   9s
build.build.openshift.io/koku-flower-1       Docker    Git@6ae7621   Complete   11 minutes ago   9s

NAME                                               DOCKER REPO                              TAGS                              UPDATED
imagestream.image.openshift.io/centos              registry.centos.org/centos/centos        latest,6,7 + 4 more...            36 minutes ago
imagestream.image.openshift.io/koku-api            172.30.1.1:5000/koku/koku-api            latest                            36 minutes ago
imagestream.image.openshift.io/koku-flower         172.30.1.1:5000/koku/koku-flower         latest                            11 minutes ago
imagestream.image.openshift.io/koku-masu-flask     172.30.1.1:5000/koku/koku-masu-flask     latest                            32 minutes ago
imagestream.image.openshift.io/koku-worker         172.30.1.1:5000/koku/koku-worker                                           
imagestream.image.openshift.io/postgresql          172.30.1.1:5000/koku/postgresql          9.6                               36 minutes ago
imagestream.image.openshift.io/python-36-centos7   172.30.1.1:5000/koku/python-36-centos7   latest                            36 minutes ago
imagestream.image.openshift.io/rabbitmq            172.30.1.1:5000/koku/rabbitmq            3.6.15                            28 minutes ago
imagestream.image.openshift.io/redis               172.30.1.1:5000/koku/redis               5.0.4                             36 minutes ago
imagestream.image.openshift.io/rhel                registry.access.redhat.com/rhel7/rhel    7.2-46,7.4,7.5-245 + 47 more...   35 minutes ago

NAME                                       HOST/PORT                                   PATH      SERVICES          PORT       TERMINATION   WILDCARD
route.route.openshift.io/koku-api          koku-api-koku.192.168.42.57.nip.io                    koku-api          <all>                    None
route.route.openshift.io/koku-flower       koku-flower-koku.192.168.42.57.nip.io                 koku-flower       <all>                    None
route.route.openshift.io/koku-masu-flask   koku-masu-flask-koku.192.168.42.57.nip.io             koku-masu-flask   8080-tcp                 None
```

The failure in the flower pod is due to an issue with the S2I build. It's unrelated to this change and will be fixed in a future PR.

### CODE REVIEW

**Comment by adberglund** [comment]
One of the acceptance criteria on the issue is that the hccm-build project does not error with the new templates. Have you looked into that?

**Comment by dccurtis** [comment]
Adding @lcouzens and @dchorvat1 so they are in the loop on this.  Any feedback or preparation comments to get them ready to update Jenkins would be very helpful.

If these templates are intended to be used before the masu/koku merge is complete we should probably put that on the acceptance criteria to make sure that still works.

**Comment by blentz** [comment]
@adberglund - I've got new jobs set up in the hccm-build project that are now passing.

File: openshift/koku-api.yaml [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+    replicas: 1
+    selector:
+      name: ${NAME}-api
+    resources:
+      requests:
+        memory: ${MEMORY_REQUEST}
+      limits:
+        memory: ${MEMORY_LIMIT}
+    strategy:
+      type: Recreate
```
Comment (adberglund) [inline openshift/koku-api.yaml]: This looks like a regression. We recently switched to rolling deployments

Comment (blentz) [reply openshift/koku-api.yaml]: Fixed! Thanks for catching that.

**Pull request merged**